Sources/NIOEmbedded/Embedded.swift

```diff
@@ -788,7 +788,8 @@ public final class EmbeddedChannel: Channel {
 
     @inlinable
     internal func getOptionSync<Option: ChannelOption>(_ option: Option) -> Option.Value {
-        if option is ChannelOptions.Types.AutoReadOption {
+        if option is ChannelOptions.Types.AutoReadOption
+        || option is ChannelOptions.Types.AllowRemoteHalfClosureOption {
```

Comment:
I know I gave you this advice, so this is really my fault, but I no longer think this behaviour is quite right. 😁 

I think we should allow the user of `EmbeddedChannel` to have a `public var` that they can set which provides the value of `AllowRemoteHalfClosureOption`. That lets tests configure whether the behaviour should be supported or not. This also lets us support `setOption` for this option.

Relatedly, the default value should be `false`.